src/modals/TransactionDetailsModal.tsx

```diff
@@ -153,16 +153,16 @@ const TransactionDetailsModal = ({ transaction, address, onClose }: TransactionD
           <DetailsRow label={t`Inputs`}>
             <AddressList>
               {transaction.inputs?.map((input) => (
-                <ActionLinkStyled key={`${input.outputRef.key}`} onClick={() => handleShowAddress(input.address)}>
-                  <AddressEllipsed key={`${input.outputRef.key}`} addressHash={input.address} />
+                <ActionLinkStyled key={`${input.outputRef.key}`} onClick={() => handleShowAddress(input.address ?? '')}>
+                  <AddressEllipsed key={`${input.outputRef.key}`} addressHash={input.address ?? ''} />
```

Comment:
Does it really make sense to render `ActionLinkStyled` and `AddressEllipsed` if `input.address` is `undefined? Instead of passing an empty string to a function and component that both expect an address hash, I would just not render the components at all.

```ts
              {transaction.inputs?.map(
                (input) =>
                  input.address && (
                    <ActionLinkStyled
                      key={`${input.outputRef.key}`}
                      onClick={() => handleShowAddress(input.address as string)}
                    >
                      <AddressEllipsed key={`${input.outputRef.key}`} addressHash={input.address} />
                    </ActionLinkStyled>
                  )
              )}
```

Reply:
They never will be undefined, but the type-checker was not happy, so this satisfied it - sure, not rendering them at all is an option too

Reply:
It seems the type-checker is really unhappy with a wrapping `input.address !== undefined && ...` ... It's been resolved using both ways